streams/src/main/java/org/apache/kafka/streams/state/internals/InMemorySessionStore.java

```diff
@@ -120,7 +120,15 @@ public void put(final Windowed<Bytes> sessionKey, final byte[] aggregate) {
     @Override
     public void remove(final Windowed<Bytes> sessionKey) {
         final ConcurrentNavigableMap<Bytes, ConcurrentNavigableMap<Long, byte[]>> keyMap = endTimeMap.get(sessionKey.window().end());
+        if (keyMap == null) {
```

Comment:
How could `keyMap` be `null`, ie, why would we call `remove()`for a non-existing session (at least in the DSL)?

Not saying that the fix does not make sense, I am just wondering if we need an additional fix for the DSL?

Reply:
Well. Thinking about it twice, during restore maybe? Tombstone are preserved longer in the changelog, hence, for a deleted session we only have the tombstone. On restore, we would never see and "insert" but only the delete.

Maybe worth double checking in the DSL code anyway.

Reply:
I assume a tombstone was being restored, where the corresponding put was already cleaned up from the changelog?

Reply:
Answered yourself quicker than me. But ok, I'll look around regardless

Reply:
I'm fairly confident the only affected case is on the restore path when the changelog contains a tombstone that outlived its corresponding "put" -- during normal processing, if the delete arrives after the key has already been cleaned up it will be dropped as a late record anyways. Assuming we aren't accidentally sending double tombstones anywhere.